test/Microsoft.Health.Fhir.Tests.E2E/Rest/BasicAuthTests.cs

```diff
@@ -104,6 +106,48 @@ async Task<FhirException> ExecuteAndValidateNotFoundStatus(Func<Task> action)
             }
         }
 
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task WhenUpdatingAResource_GivenAUserWithUpdatePermissions_TheServerShouldReturnSuccess()
+        {
+            await Client.RunAsUser(TestUsers.AdminUser, TestApplications.NativeClient);
+            Observation createdResource = await Client.CreateAsync(Samples.GetDefaultObservation());
+
+            await Client.RunAsUser(TestUsers.ReadWriteUser, TestApplications.NativeClient);
+            FhirResponse<Observation> updateResponse = await Client.UpdateAsync(createdResource);
+
+            Assert.Equal(System.Net.HttpStatusCode.OK, updateResponse.StatusCode);
+
+            Observation updatedResource = updateResponse.Resource;
+
+            Assert.NotNull(updatedResource);
+            Assert.Equal(createdResource.Id, updatedResource.Id);
+            Assert.NotEqual(createdResource.Meta.VersionId, updatedResource.Meta.VersionId);
+            Assert.NotEqual(createdResource.Meta.LastUpdated, updatedResource.Meta.LastUpdated);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task WhenGettingAResource_GivenAClientWithNoAuthToken_TheServerShouldReturnUnAuthorized()
+        {
+            await Client.RunAsClientApplication(TestApplications.InvalidClient);
+
+            FhirException fhirException = await Assert.ThrowsAsync<FhirException>(async () => await Client.CreateAsync(Samples.GetDefaultObservation()));
```

Comment:
Test method name is WhenGetting, but we're creating here. Same thing in other new test methods.